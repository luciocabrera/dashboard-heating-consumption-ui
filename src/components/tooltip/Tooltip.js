// React
import React from 'react';
// Components
import { Popover } from 'antd';
// Css
import './tooltip.css';

const Tooltip = (props) => {
  const { content, children } = props;
  const key = Math.random().toString(36).slice(-10);

  return (
    <Popover key={`rc-tooltip-${key}`} content={content}>
      {children}
    </Popover>
  );
};

export default Tooltip;
