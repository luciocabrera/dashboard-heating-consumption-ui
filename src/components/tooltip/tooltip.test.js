// /* globals describe, it, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Tooltip from './Tooltip';

const getProps = () => ({
  content: 'this is a tooltiptext',
  children:<div>this is a child</div>
});

describe('Rendering Tooltip', () => {
  it('Tooltip renders using ReactDOM without crashing', () => {
    const props = getProps();
    const div = document.createElement('div');
    ReactDOM.render(<Tooltip {...props} />, div);
  });
  it('Tooltip renders using enzyme shallow without crashing', () => {
    const props = getProps();
    shallow(<Tooltip {...props} />);
  });
  it('Tooltip renders using enzyme mount without crashing', () => {
    const props = getProps();
    const wrapper = mount(<Tooltip {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
});

describe('Tooltip (Snapshot)', () => {
  it('Tooltip renders using renderer without crashing', () => {
    const props = getProps();
    const component = renderer.create(<Tooltip {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Tooltip', () => {
  it('should show children', () => {
    const props = getProps();
    const component = mount(<Tooltip {...props} />);
    const actual = component.html();
    const expected = `<div>this is a child</div>`;
    expect(actual).toContain(expected);
  });
});
