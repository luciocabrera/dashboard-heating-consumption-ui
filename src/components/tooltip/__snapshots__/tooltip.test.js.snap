// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip (Snapshot) Tooltip renders using renderer without crashing 1`] = `
<div
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  this is a child
</div>
`;
